rls/schemas.py

```diff
@@ -17,115 +19,83 @@ class Command(str, Enum):
+class ConditionArg(BaseModel):
     comparator_name: str
     type: Type[sqltypes.TypeEngine]
-    operation: NotRequired[Optional[Operation]]
-    column_name: NotRequired[Optional[str]]
 
 
 class Policy(BaseModel):
     definition: str
-    condition_args: List[ConditionArgs]
+    condition_args: List[ConditionArg]
     cmd: Union[Command, List[Command]]
-    joined_expr: Optional[str] = None
-    custom_expr: Optional[str] = None
+    custom_expr: Callable[..., ClauseElement]
 
     __policy_names: List[str] = []
     __expr: str = ""
     __policy_suffix: str = ""
+    __condition_args_prefix: str = "rls"
 
     class Config:
         arbitrary_types_allowed = True
 
-    def _get_safe_variable_name(self, idx: int = 0):
-        type_str = self.condition_args[idx]["type"].__visit_name__.upper()
-        return f"NULLIF(current_setting('rls.{self.condition_args[idx]["comparator_name"]}', true),'')::{type_str}"
-
-    def _get_expr_from_params(self, table_name: str, idx: int = 0):
-        safe_variable_name = self._get_safe_variable_name(idx=idx)
-
-        operation_obj = self.condition_args[idx]
-
-        # Check if "operation" exists and is not None before accessing its value
-        if "operation" in operation_obj and operation_obj["operation"] is not None:
-            operation_value = operation_obj["operation"].value
-        else:
-            operation_value = ""
-
-        expr = f"{self.condition_args[idx]['column_name']} {operation_value} {safe_variable_name}"
+    def _ensure_boolean(self, expression: ClauseElement):
+        """
+        Ensures that the given expression evaluates to a Boolean value.
+        If not, it casts the expression to Boolean.
+        """
+        # Check if the expression is already of Boolean type
+        if isinstance(expression.type, Boolean):
+            return expression
+
+        # Otherwise, cast the expression to Boolean explicitly or raise an error
+        raise ValueError("Expression does not evaluate to a Boolean value")
+        # return expression.cast(Boolean)
+
+    def _validate_Arguments_length(self):
+        condition_args_length = len(self.condition_args)
+        lamda_args_length = len(inspect.signature(self.custom_expr).parameters)
+        if condition_args_length != lamda_args_length:
+            raise ValueError(
+                f"Length mismatch for arguments. Expected {condition_args_length}, got {lamda_args_length}"
+            )
+        return True
+
+    def _convert_lambda_to_clause_element(self):
+        """Convert the lambda function to a SQLAlchemy expression."""
+        args = []
+        for arg in self.condition_args:
+            coalesced_value = func.coalesce(
+                func.current_setting(
+                    f"{self.__condition_args_prefix}.{arg.comparator_name}"
+                ),
+                "",
+            ).cast(arg.type)
+            args.append(coalesced_value)
+        self.custom_expr = self.custom_expr(*args)
```

Comment:
we should not overwrite this, maybe we do `self.compiled_expr = self.custom_expr(*args)` instead?